xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-api/src/main/java/org/xwiki/rendering/macro/wikibridge/WikiMacroParameterDescriptor.java

```diff
@@ -134,16 +134,11 @@ public String getDescription()
     @Deprecated
     public Class< ? > getType()
     {
-        Class<?> clazz;
-        if (this.parameterType instanceof Class) {
-            clazz = (Class) this.parameterType;
-        } else if (this.parameterType instanceof ParameterizedType) {
-            clazz = (Class) ((ParameterizedType) this.parameterType).getRawType();
+        if (this.parameterType == null) {
```

Comment:
I would change a bit the logic and always call ReflectionUtils.getTypeClass but check the return for null and return String in that case.